From: Kill Your TV <killyourtv@i2pmail.org>
Date: Thu, 28 Aug 2014 09:26:00 +0000
Subject: ldflags

Add the LDFLAGS variable to the makefiles so packages can be hardened without
the obsolete hardening-wrapper package.
---
 src/c/Makefile-linux-armel-32.make | 4 ++--
 src/c/Makefile-linux-armhf-32.make | 4 ++--
 src/c/Makefile-linux-ia-64.make    | 4 ++--
 src/c/Makefile-linux-ppc-32.make   | 4 ++--
 src/c/Makefile-linux-ppc-64.make   | 4 ++--
 src/c/Makefile-linux-x86-32.make   | 6 +++---
 src/c/Makefile-linux-x86-64.make   | 6 +++---
 7 files changed, 16 insertions(+), 16 deletions(-)

diff --git a/src/c/Makefile-linux-armel-32.make b/src/c/Makefile-linux-armel-32.make
index c257910..fd99ec7 100644
--- a/src/c/Makefile-linux-armel-32.make
+++ b/src/c/Makefile-linux-armel-32.make
@@ -33,10 +33,10 @@ init:
 	if test ! -d .deps; then mkdir .deps; fi
 
 wrapper: $(wrapper_SOURCE)
-	$(COMPILE) -pthread $(wrapper_SOURCE) -lm -o $(BIN)/wrapper
+	$(COMPILE) -pthread $(wrapper_SOURCE) -lm $(LDFLAGS) -o $(BIN)/wrapper
 
 libwrapper.so: $(libwrapper_so_OBJECTS)
-	${COMPILE} -shared $(libwrapper_so_OBJECTS) -o $(LIB)/libwrapper.so
+	${COMPILE} -shared $(libwrapper_so_OBJECTS) $(LDFLAGS) -o $(LIB)/libwrapper.so
 
 %.o: %.c
 	@echo '$(COMPILE) -c $<'; \
diff --git a/src/c/Makefile-linux-armhf-32.make b/src/c/Makefile-linux-armhf-32.make
index c257910..fd99ec7 100644
--- a/src/c/Makefile-linux-armhf-32.make
+++ b/src/c/Makefile-linux-armhf-32.make
@@ -33,10 +33,10 @@ init:
 	if test ! -d .deps; then mkdir .deps; fi
 
 wrapper: $(wrapper_SOURCE)
-	$(COMPILE) -pthread $(wrapper_SOURCE) -lm -o $(BIN)/wrapper
+	$(COMPILE) -pthread $(wrapper_SOURCE) -lm $(LDFLAGS) -o $(BIN)/wrapper
 
 libwrapper.so: $(libwrapper_so_OBJECTS)
-	${COMPILE} -shared $(libwrapper_so_OBJECTS) -o $(LIB)/libwrapper.so
+	${COMPILE} -shared $(libwrapper_so_OBJECTS) $(LDFLAGS) -o $(LIB)/libwrapper.so
 
 %.o: %.c
 	@echo '$(COMPILE) -c $<'; \
diff --git a/src/c/Makefile-linux-ia-64.make b/src/c/Makefile-linux-ia-64.make
index d4be3f1..4ec89ff 100644
--- a/src/c/Makefile-linux-ia-64.make
+++ b/src/c/Makefile-linux-ia-64.make
@@ -33,10 +33,10 @@ init:
 	if test ! -d .deps; then mkdir .deps; fi
 
 wrapper: $(wrapper_SOURCE)
-	$(COMPILE) -lm -pthread $(wrapper_SOURCE) -o $(BIN)/wrapper
+	$(COMPILE) -lm -pthread $(LDFLAGS) $(wrapper_SOURCE) -o $(BIN)/wrapper
 
 libwrapper.so: $(libwrapper_so_OBJECTS)
-	${COMPILE} -lm -shared $(libwrapper_so_OBJECTS) -o $(LIB)/libwrapper.so
+	${COMPILE} -lm -shared $(LDFLAGS) $(libwrapper_so_OBJECTS) -o $(LIB)/libwrapper.so
 
 %.o: %.c
 	@echo '$(COMPILE) -c $<'; \
diff --git a/src/c/Makefile-linux-ppc-32.make b/src/c/Makefile-linux-ppc-32.make
index fbc2424..8dd57c4 100644
--- a/src/c/Makefile-linux-ppc-32.make
+++ b/src/c/Makefile-linux-ppc-32.make
@@ -33,10 +33,10 @@ init:
 	if test ! -d .deps; then mkdir .deps; fi
 
 wrapper: $(wrapper_SOURCE)
-	$(COMPILE) -lm -pthread $(wrapper_SOURCE) -o $(BIN)/wrapper
+	$(COMPILE) -lm -pthread $(LDFLAGS) $(wrapper_SOURCE) -o $(BIN)/wrapper
 
 libwrapper.so: $(libwrapper_so_OBJECTS)
-	${COMPILE} -shared $(libwrapper_so_OBJECTS) -o $(LIB)/libwrapper.so
+	${COMPILE} -shared $(LDFLAGS) $(libwrapper_so_OBJECTS) -o $(LIB)/libwrapper.so
 
 %.o: %.c
 	@echo '$(COMPILE) -c $<'; \
diff --git a/src/c/Makefile-linux-ppc-64.make b/src/c/Makefile-linux-ppc-64.make
index 78db926..f32b2a3 100644
--- a/src/c/Makefile-linux-ppc-64.make
+++ b/src/c/Makefile-linux-ppc-64.make
@@ -33,10 +33,10 @@ init:
 	if test ! -d .deps; then mkdir .deps; fi
 
 wrapper: $(wrapper_SOURCE)
-	$(COMPILE) -lm -pthread $(wrapper_SOURCE) -o $(BIN)/wrapper
+	$(COMPILE) -lm -pthread $(LDFLAGS) $(wrapper_SOURCE) -o $(BIN)/wrapper
 
 libwrapper.so: $(libwrapper_so_OBJECTS)
-	${COMPILE} -shared $(libwrapper_so_OBJECTS) -o $(LIB)/libwrapper.so
+	${COMPILE} -shared $(LDFLAGS) $(libwrapper_so_OBJECTS) -o $(LIB)/libwrapper.so
 
 %.o: %.c
 	@echo '$(COMPILE) -c $<'; \
diff --git a/src/c/Makefile-linux-x86-32.make b/src/c/Makefile-linux-x86-32.make
index 5cad095..f5697c2 100644
--- a/src/c/Makefile-linux-x86-32.make
+++ b/src/c/Makefile-linux-x86-32.make
@@ -36,13 +36,13 @@ init:
 	if test ! -d .deps; then mkdir .deps; fi
 
 wrapper: $(wrapper_SOURCE)
-	$(COMPILE) -pthread $(wrapper_SOURCE) -lm -o $(BIN)/wrapper
+	$(COMPILE) -pthread $(wrapper_SOURCE) -lm $(LDFLAGS) -o $(BIN)/wrapper
 
 testsuite: $(testsuite_SOURCE)
-	$(COMPILE) -DCUNIT $(testsuite_SOURCE) -lm -pthread -L/usr/local/lib -lncurses -lcunit -o $(TEST)/testsuite
+	$(COMPILE) -DCUNIT $(testsuite_SOURCE) -lm $(LDFLAGS) -pthread -L/usr/local/lib -lncurses -lcunit -o $(TEST)/testsuite
 
 libwrapper.so: $(libwrapper_so_OBJECTS)
-	${COMPILE} -shared $(libwrapper_so_OBJECTS) -o $(LIB)/libwrapper.so
+	${COMPILE} -shared $(libwrapper_so_OBJECTS) $(LDFLAGS) -o $(LIB)/libwrapper.so
 
 %.o: %.c
 	@echo '$(COMPILE) -c $<'; \
diff --git a/src/c/Makefile-linux-x86-64.make b/src/c/Makefile-linux-x86-64.make
index d6af907..cab1d9f 100644
--- a/src/c/Makefile-linux-x86-64.make
+++ b/src/c/Makefile-linux-x86-64.make
@@ -36,13 +36,13 @@ init:
 	if test ! -d .deps; then mkdir .deps; fi
 
 wrapper: $(wrapper_SOURCE)
-	$(COMPILE) -pthread $(wrapper_SOURCE) -lm -o $(BIN)/wrapper
+	$(COMPILE) -pthread $(wrapper_SOURCE) -lm $(LDFLAGS) -o $(BIN)/wrapper
 
 testsuite: $(testsuite_SOURCE)
-	$(COMPILE) -DCUNIT $(testsuite_SOURCE) -lm -pthread -L/usr/local/lib -lncurses -lcunit -o $(TEST)/testsuite
+	$(COMPILE) -DCUNIT $(testsuite_SOURCE) -lm $(LDFLAGS) -pthread -L/usr/local/lib -lncurses -lcunit -o $(TEST)/testsuite
 
 libwrapper.so: $(libwrapper_so_OBJECTS)
-	${COMPILE} -shared $(libwrapper_so_OBJECTS) -o $(LIB)/libwrapper.so
+	${COMPILE} -shared $(libwrapper_so_OBJECTS) $(LDFLAGS) -o $(LIB)/libwrapper.so
 
 %.o: %.c 
 	@echo '$(COMPILE) -c $<'; \
